src/TextField/TextFieldLabel.js

```diff
@@ -35,13 +43,30 @@ export default function TextFieldLabel(props, context) {
   }, classNameProp);
 
   return (
-    <label className={className} {...other} />
+    <label className={className} {...other}>
+      {children}
+      {required ? <span className={asteriskClassName}>{'\u2009'}*</span> : null}
```

Comment:
Regarding the thin space: It really looks better and more like in the specs with a thin space! :smile: